File: src/main/java/com/rarmash/cs2_simulator/skinspecs/WeaponType.java
package com.rarmash.cs2_simulator.skinspecs;

public enum WeaponType {
    PISTOL ("Pistol"),
    SMG ("SMG"),
    SNIPER_RIFLE ("Sniper rifle"),
    RIFLE ("Rifle"),
    KNIFE ("Knife"),
    SHOTGUN ("Shotgun"),
    MACHINE_GUN ("Machine gun"),
    GLOVES ("Gloves"),
    EQUIPMENT ("Equipment");

    private String weaponType;

    WeaponType(String weaponTypeString) {
        this.weaponType = weaponTypeString;
    }

    public String getWeaponTypeString() {
        return weaponType;
    }

    public static WeaponType fromString(String weaponTypeString) {
        for (WeaponType weapon : WeaponType.values()) {
            if (weapon.getWeaponTypeString().equalsIgnoreCase(weaponTypeString)) {
                return weapon;
            }
        }
        throw new IllegalArgumentException("No enum constant for weapon type title: " + weaponTypeString);
    }

    @Override
    public String toString() {
        return "WeaponType {" +
                "weaponType='" + weaponType + '\'' +
                '}';
    }
}

File: src/main/kotlin/com/rarmash/cs2_simulator/model/enums/WeaponType.kt
package com.rarmash.cs2_simulator.model.enums

enum class WeaponType(val weaponType: String) {
    PISTOL("Pistol"),
    SMG("SMG"),
    SNIPER_RIFLE("Sniper Rifle"),
    RIFLE("Rifle"),
    KNIFE("Knife"),
    SHOTGUN("Shotgun"),
    MACHINE_GUN("Machine Gun"),
    GLOVES("Gloves"),
    EQUIPMENT("Equipment");

    companion object {
        private val map = entries.associateBy { it.weaponType.lowercase() }

        @JvmStatic
        fun fromString(weaponTypeString: String): WeaponType {
            return map[weaponTypeString.lowercase()]
                ?: throw IllegalArgumentException("No enum constant for weapon type title: $weaponTypeString")
        }
    }

    override fun toString(): String = weaponType
}